apiroutes/templates.go

```diff
@@ -0,0 +1,192 @@
+/*******************************************************************************
+ * Copyright (c) 2019 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+
+package apiroutes
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/eclipse/codewind-installer/config"
+)
+
+type (
+	// Template represents a project template.
+	Template struct {
+		Label       string `json:"label"`
+		Description string `json:"description"`
+		Language    string `json:"language"`
+		URL         string `json:"url"`
+		ProjectType string `json:"projectType"`
+	}
+
+	// TemplateRepo represents a template repository.
+	TemplateRepo struct {
+		Description string `json:"description"`
+		URL         string `json:"url"`
+	}
+)
+
+// GetTemplates gets project templates from PFE's REST API.
+// Filter them using the function arguments
+func GetTemplates(projectStyle string, showEnabledOnly string) ([]Template, error) {
+	req, err := http.NewRequest("GET", config.PFEApiRoute+"templates", nil)
```

Comment:
```suggestion
	req, err := http.NewRequest("GET", config.PFEApiRoute + "templates", nil)
```

Reply:
I thought this but the Go formatter automatically removed the spaces